On april fools mothpeople are replaced by monthpeople

tralezab opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Replaces mothpeople with monthpeople on april fools. they're living calendars!


They will have a check to not be available to admins either so i'm adding another check to not even make these admin available, I don't want them to ruin the feature. This should be completely dormant because this feature is retarded.

TODO:

- [x] eye organ in the head moved to chest
- [x] wearing jumpsuits without adjusting them blinds you because you zip it up over your head
- [x] eye colors
- [x] ~mutcolors for the red bit?~ eh

make as much as i can wearable, that is:
- [x] hats
- [x] radios
- [x] ~glasses~ i tried, glasses don't fit at all
- [x] any way to do backpacks better

I only have the downwards directional sprites at the moment because i'm still unsure of how I want them to look. i'm still editting it a lot!

## Why It's Good For The Game

April fools is silly and I really like the idea that 1/5 of the station consists of cartoonish stubby sentient calendars, might give them the shrink component too so they can be tiny

current iteration: atmos techie monthperson works hard to keep you breathing out of the bottom of the heart it doesn't have because it doesn't need the air, you do
![image](https://user-images.githubusercontent.com/40974010/63630137-83fe6b80-c5cc-11e9-9377-09beeeb817eb.png)

![image](https://user-images.githubusercontent.com/40974010/63646647-efc6fe00-c6ca-11e9-9a36-99c4c5239b60.png)

tralezab committed:
monthpeople (moth removal on april fools)

tralezab committed:
random names, bad paths fixed, on species loss added, sprite changes

tralezab commented:
<details>
  <summary>A very bad no good picture from fikou of monthpeople</summary>

  ![image](https://user-images.githubusercontent.com/40974010/63629327-63cbae00-c5c6-11e9-95ea-eaecc301732b.png)

</details>

tralezab commented on code/modules/mob/living/carbon/human/species_types/monthmen.dm:
```diff
@@ -0,0 +1,74 @@
+	species_traits = list(NO_UNDERWEAR, NOBLOOD)
+	default_features = list("mcolor" = "FFF", "tail_human" = "None", "ears" = "None", "wings" = "None")
+	skinned_type = /obj/item/paper
+	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_PRIDE | MIRROR_MAGIC | ERT_SPAWN
```
almost all of these are going to be removed

tralezab replied:
done

tralezab committed:
probably the final look of the monthpeople

Fikou commented:
theres someone that lets you place glasses in a custom place its used on snails

tralezab commented:
that's what i intend to do, though i'm gonna have to think of a way to let them equip those items without a head

tralezab committed:
100% removes any way to get monthpeople, small glove changes

PKPenguin321 commented:
Abstract head body part? :V

tralezab commented:
😅 we'll try it!!

I got most of the beginning stuff I wanted done so i'm gonna start on that checklist

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/monthmen.dm:
```diff
@@ -0,0 +1,74 @@
+/datum/species/monthmen
+	//an exotic species that arrives once a year to remove the worst species, mothpeople.
```
I'LL HAVE YOU KNOW-

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/monthmen.dm:
```diff
@@ -0,0 +1,74 @@
+	mutanttongue = /obj/item/organ/tongue/monthmen
+	mutantears = /obj/item/organ/ears/monthmen
+	//other traits
+	siemens_coeff = 0 //not very good at conducting electricity
```
👀

tralezab replied:
correct me if i'm wrong...

carlarctg replied:
power creep

tralezab committed:
headsets finished

tralezab committed:
fixes species dropdown menu

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into monthpeople

tralezab committed:
compiles + uses NONE for monthmen changesource

tralezab commented:
listen nobody cares at all but with this new abstract head system we could get dullahans to wear hats and lipstick

tralezab committed:
different meat for monthpeople, updates unit test, start of head stuff

tralezab committed:
yet another change for the unit test (eligeab

tralezab commented on code/modules/unit_tests/species_whitelists.dm:
```diff
@@ -1,5 +1,5 @@
-		if(initial(S.changesource_flags) == NONE)
-			Fail("A species type was detected with no changesource flags: [S]")
+		if(initial(S.changesource_flags) == NONE && S.check_roundstart_eligible() == IMPOSSIBLE) //cannot be made in any way + cannot be roundstart = completely unavailable
+			Fail("A species type was detected with no changesource flags, and is not roundstart eligible: [S]")
```
@optimumtact this alright? by default, check_roundstart_eligible will give IMPOSSIBLE if it's not enabled roundstart, but children can make the argument POSSIBLE which still passes the check, AKA it's not roundstart and cannot be obtained but it is enabled in an event, so it's OK.

optimumtact replied:
why are you doing it this way, just make another changesource flag that has no direct on station equivalent and call it EVENTRACE

tralezab replied:
I like it, I'll do that

SpaceManiac replied:
this is trying to call a proc on a typepath

tralezab commented on code/modules/mob/living/carbon/human/species_types/monthmen.dm:
```diff
@@ -0,0 +1,76 @@
+	species_traits = list(NO_UNDERWEAR, NOBLOOD, ABSTRACT_HEAD)
+	default_features = list("mcolor" = "FFF", "tail_human" = "None", "ears" = "None", "wings" = "None")
+	skinned_type = /obj/item/paper
+	changesource_flags = NONE //absolutely no way to get the race, it's completely out of theme of ss13 and should stay locked up in april fools
```
Travis will fail with this pr until it's merged because it fixes the unit test in the same pr that fails the unit test

qwertyquerty replied:
Pretty sure it uses the unit test of the current PR

tralezab committed:
smol commit

qwertyquerty commented on code/__DEFINES/mobs.dm:
```diff
@@ -291,8 +291,8 @@
-//Badmin magic mirror
-#define MIRROR_BADMIN (1<<0)
+//Badmin magic mirror, VV dropdown menu. Basically things that only admins could spawn.
+#define ADMIN (1<<0)
```
Not sure why you changed this but I would at least keep it as MIRROR_ADMIN

SpaceManiac replied:
agree

qwertyquerty commented on code/modules/mob/living/carbon/human/species_types/shadowpeople.dm:
```diff
@@ -61,7 +61,7 @@
 
 /datum/species/shadow/nightmare/check_roundstart_eligible()
-	return FALSE
+	return .....()//skips the parent proc so is just the normal roundstart eligible (don't pass POSSIBLE)
```
If you're trying to call species/check_roundstart_eligible I believe this should just be ...()

As well as for all the others

SpaceManiac replied:
Do not `.....()`

optimumtact replied:
what in the everloving hell is this, if you ever write this line of code it is a sign you are doing something massively stupid.

Add an override var

bgobandit commented:
this is the dumbest shit I have ever seen

I love it

tralezab commented:
thanks for the help everyone

tralezab committed:
Revert "yet another change for the unit test (eligeab"

This reverts commit b4da8cb5c6e19da3407f83194576b9dba716b1ab.

tralezab committed:
for real finishes the unit test insanity

tralezab committed:
whoops

tralezab committed:
eyes moved to chest from abstract 1/2

tralezab committed:
eyes moved to chest from abstract 2/2 + other goodies

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into monthpeople

tralezab committed:
fixes set species, removes nojumpsuit

NEXT: eyeball removal problem

tralezab committed:
eyeball sprite change

tralezab commented:
gonna be away for a couple days, will return the moment i'm back

tralezab commented:
im back

SpaceManiac commented:
are you though

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into monthpeople

tralezab commented:
it's so close to being done i have no excuse

lets get that monthly bread

tralezab committed:
bad define fixed

tralezab committed:
one last define swap

tralezab committed:
fixes the eyeball problem

tralezab committed:
removes debugging messages

tralezab committed:
covered chest = blind af

tralezab committed:
finishing the rest of the sprites 1/2

easy ones

tralezab commented:
![image](https://user-images.githubusercontent.com/40974010/64491977-631c5400-d223-11e9-8443-da43b9dd272d.png)
he can't see!
![image](https://user-images.githubusercontent.com/40974010/64491986-7fb88c00-d223-11e9-95dd-54a043362088.png)
now he can! (clothes blocking sight done)

nemvar commented on code/modules/mob/living/carbon/human/species_types/monthmen.dm:
```diff
@@ -0,0 +1,102 @@
+	var/days_in_that_month = 31
+	switch(month)
+		if("February")//hey look the FUCK month where i have to calculate leap years
+			if(isLeap(text2num(time2text(world.timeofday, "YY"))))
```
![wdada](https://user-images.githubusercontent.com/47324920/64492059-ee233b80-d26f-11e9-9fdc-6db90329d48e.PNG)

tralezab replied:
calculating leap years in random_name was lame anyways, lets just put that on the table

tralezab committed:
finishing the rest of sprites 2/2 + more

removes debug spawn flag, moves some messages around, removes leap year calculation

tralezab committed:
single commit for restricting monthpeople further

tralezab commented:
dead again RIP

optimumtact commented:
:spiral_calendar:
